File: gto-support-appcompat/src/main/java/androidx/appcompat/widget/AppCompatButtonImpl.java
package androidx.appcompat.widget;

import android.content.Context;
import android.util.AttributeSet;

import org.ccci.gto.android.common.appcompat.R;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import static androidx.appcompat.widget.ReflectionUtils.getDeclaredField;

public class AppCompatButtonImpl extends AppCompatButton {
    private final AppCompatCompoundDrawableHelper mCompoundDrawableHelper;

    public AppCompatButtonImpl(@NonNull final Context context) {
        this(context, null);
    }

    public AppCompatButtonImpl(@NonNull final Context context, @Nullable final AttributeSet attrs) {
        this(context, attrs, R.attr.buttonStyle);
    }

    public AppCompatButtonImpl(@NonNull final Context context, @Nullable final AttributeSet attrs,
                               final int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        mCompoundDrawableHelper =
                new AppCompatCompoundDrawableHelper(this, getDeclaredField(AppCompatButton.class, "mTextHelper"));
        mCompoundDrawableHelper.loadFromAttributes(attrs, defStyleAttr);
    }

    @Override
    protected void drawableStateChanged() {
        super.drawableStateChanged();
        if (mCompoundDrawableHelper != null) {
            mCompoundDrawableHelper.applyCompoundDrawablesTints();
        }
    }
}

File: gto-support-appcompat/src/main/java/androidx/appcompat/widget/AppCompatButtonImpl.kt
package androidx.appcompat.widget

import android.content.Context
import android.util.AttributeSet
import org.ccci.gto.android.common.appcompat.R
import org.ccci.gto.android.common.util.getDeclaredFieldOrNull

private val textHelperField by lazy { getDeclaredFieldOrNull<AppCompatButton>("mTextHelper") }

class AppCompatButtonImpl @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = R.attr.buttonStyle
) : AppCompatButton(context, attrs, defStyleAttr) {
    private val compoundDrawableHelper: AppCompatCompoundDrawableHelper?

    init {
        compoundDrawableHelper = AppCompatCompoundDrawableHelper(this, textHelperField)
        compoundDrawableHelper.loadFromAttributes(attrs, defStyleAttr)
    }

    override fun drawableStateChanged() {
        super.drawableStateChanged()
        compoundDrawableHelper?.applyCompoundDrawablesTints()
    }
}
